cli/tools/repl.rs

```diff
@@ -50,56 +50,42 @@ impl EditorHelper {
     self.message_tx.send((method.to_string(), params))?;
     self.response_rx.recv()?
   }
-}
-
-fn is_word_boundary(c: char) -> bool {
-  if c == '.' {
-    false
-  } else {
-    char::is_ascii_whitespace(&c) || char::is_ascii_punctuation(&c)
-  }
-}
-
-impl Completer for EditorHelper {
-  type Candidate = String;
 
-  fn complete(
-    &self,
-    line: &str,
-    pos: usize,
-    _ctx: &Context<'_>,
-  ) -> Result<(usize, Vec<String>), ReadlineError> {
-    let start = line[..pos].rfind(is_word_boundary).map_or_else(|| 0, |i| i);
-    let end = line[pos..]
-      .rfind(is_word_boundary)
-      .map_or_else(|| pos, |i| pos + i);
-
-    let word = &line[start..end];
-    let word = word.strip_prefix(is_word_boundary).unwrap_or(word);
-    let word = word.strip_suffix(is_word_boundary).unwrap_or(word);
-
-    let fallback = format!(".{}", word);
+  fn get_global_lexical_scope_names(&self) -> Vec<String> {
+    let evaluate_response = self
+      .post_message(
+        "Runtime.globalLexicalScopeNames",
```

Comment:
.@caspervonb gave me the tip here: https://chromedevtools.github.io/devtools-protocol/v8/Runtime/#method-globalLexicalScopeNames